lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -273,6 +273,9 @@ sub _prepare_complex_query_search_args ($self, $args) {
         my $subquery = $schema->resultset('JobGroups')->search({name => $args->{group}})->get_column('id')->as_query;
         push @conds, {'me.group_id' => {-in => $subquery}};
     }
+    if (defined $args->{not_groupid}) {
+        push @conds, {'me.group_id' => {-not_in => $args->{not_groupid}}};
+    }
```

Comment:
How about this:
```suggestion
    if (defined $args->{not_groupid}) {
        my $id = $args->{not_groupid;
        if ($id) {
            push @conds, {-or => [
                {'me.group_id' => {-not_in => $args->{not_groupid}}},
                {'me.group_id' => {-not => undef}},
                ]};
        }
        else {
            push @conds, {'me.group_id' => {-not => undef}};
        }
    }
```
not tested, so check the generated SQL.

Reply:
thank you, this lets my test pass :) :+1: 
Not sure if i'm doing this right, i'm trying to have a look at the queries using `export DBI_TRACE=1;...` but it does not show a statement with the group_id value in the condition anywhere. I only find it after a INSERT and some SELECT statements

Reply:
DBI_TRACE should show you everything. 
mind that the shown SQL statements are using placeholders, with the actual values listed after it

Reply:
`DBIC_TRACE` is alot better readable, thanks. Tried again to [find the query](https://drive.zoopa.dev/code/#/2/code/view/2BZBotsLtFFFE-ibOMjz3fBJOLTXNWFgUtNu4LzaK94/embed/present/) with grep but havent found it yet

Reply:
I can see it:
```
SELECT me.id, me.result_dir, me.archived, me.state, me.priority, me.result, me.reason, me.clone_id, me.blocked_
by_id, me.TEST, me.DISTRI, me.VERSION, me.FLAVOR, me.ARCH, me.BUILD, me.MACHINE, me.group_id, me.assigned_worke
r_id, me.t_started, me.t_finished, me.logs_present, me.passed_module_count, me.failed_module_count, me.softfail
ed_module_count, me.skipped_module_count, me.externally_skipped_module_count, me.scheduled_product_id, me.resul
t_size, me.t_created, me.t_updated FROM jobs me WHERE ( ( me.group_id NOT IN ( ? ) OR me.group_id IS NULL ) ) O
RDER BY me.id DESC LIMIT ?: '1001', '10001'
```